mirgecom/initializers.py

```diff
@@ -840,3 +841,149 @@ def __call__(self, x_vec, eos, *, t=0.0):
+
+        Parameters
+        ----------
+        dim: int
+            specifies the number of dimensions for the solution
+        x0: float
+           location of discontinuity
+        nspeces: int
+            specifies the number of mixture species
+        pl: float
+            pressure to the left of the discontinuity
+        tl: float
+            temperature to the left of the discontinuity
+        ul: numpy.ndarray
+            velocity (vector) to the left of the discontinuity
+        yl: numpy.ndarray
+            species mass fractions to the left of the discontinuity
+        pr: float
+            pressure to the right of the discontinuity
+        tr: float
+            temperaure to the right of the discontinuity
+        ur: numpy.ndarray
+            velocity (vector) to the right of the discontinuity
+        yr: numpy.ndarray
+            species mass fractions to the right of the discontinuity
+        sigma: float
+           sharpness parameter
+        uc: numpy.ndarray
+            convective velocity (discontinuity advection speed)
+        """
+        if ul is None:
+            ul = np.zeros(shape=(dim,))
+        if ur is None:
+            ur = np.zeros(shape=(dim,))
+
+        if yl is None:
+            yl = np.zeros(shape=(nspecies,))
+        if yr is None:
+            yr = np.zeros(shape=(nspecies,))
+
+        if uc is None:
+            uc = np.zeros(shape=(dim,))
+
+        self._nspecies = nspecies
+        self._dim = dim
+        self._x0 = x0
+        self._sigma = sigma
+        self._ul = ul
+        self._ur = ur
+        self._uc = uc
+        self._pl = pl
+        self._pr = pr
+        self._tl = tl
+        self._tr = tr
+        self._yl = yl
+        self._yr = yr
+        self._xdir = xdir
+        if self._xdir >= self._dim:
+            self._xdir = self._dim - 1
+
+    def __call__(self, x_vec, eos, *, t=0.0):
+        """
+        Create the mixture state at locations *x_vec* (t is ignored).
+
+        Parameters
+        ----------
+        x_vec: numpy.ndarray
+            Coordinates at which solution is desired
+        eos:
+            Mixture-compatible equation-of-state object must provide
+            these functions:
+            `eos.get_density`
+            `eos.get_internal_energy`
+        t: float
+            Time is ignored by this solution intitializer
+        """
+        if x_vec.shape != (self._dim,):
+            raise ValueError(f"Position vector has unexpected dimensionality,"
+                             f" expected {self._dim}.")
+
+        x_rel = x_vec[self._xdir]
+        actx = x_rel.array_context
+        zeros = 0*x_rel
+        x0 = zeros + self._uc[self._xdir]*t + self._x0
+        ones = zeros + 1
+
+        pl = self._pl*ones
+        tl = self._tl*ones
+        ul = make_obj_array([self._ul[i]*ones for i in range(self._dim)])
+        yl = make_obj_array([self._yl[i]*ones for i in range(self._nspecies)])
```

Comment:
Real question: Are these needed? Or would something like this suffice:
```
ul = self._ul * ones
```

Reply:
Actually, it doesn't seem like you would need these arrays at all, maybe? I think the expressions below should work even if `ul`, `ur`, etc. are just floats.